### PR TITLE
Apple TV support 2: add TV template, fix config plugin, add CI

### PR DESCRIPTION
**Pull request by douglowder**
# Why

Second of a series of PRs to add Apple TV support for many (but not all) Expo SDK modules.

# How

- Add new template, `expo-template-tv`, for use with `create-expo-app`
- Fix `DeviceFamily.ts` in `@expo/config-plugins` to detect when an Xcode build configuration is for TV, and set the device family correctly
- Add small modifications to the Updates E2E scripts to create an EAS project configured for TV
- Add new option to the Updates E2E github workflow to run the E2E build on Apple TV (build only, no actual Detox tests)

This PR is broken up into a series of logically separate commits for easier reviewing.

# Test Plan

- New updates E2E job (build only) for Apple TV should pass
- All other CI should be unaffected and should pass

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by sunzongzheng** [comment]
Can I try it out before MR merge? @douglowder

**Comment by douglowder** [comment]
> Can I try it out before MR merge?

@sunzongzheng until we actually publish alpha packages with these changes, trying this out would involve pulling the entire monorepo and creating your own packages, and be very difficult and awkward.

File: packages/expo-updates/e2e/fixtures/project_files/eas-hooks/eas-build-pre-install.sh
Comment (wschurman) [inline packages/expo-updates/e2e/fixtures/project_files/eas-hooks/eas-build-pre-install.sh:42]: uncomment or remove

File: packages/expo-updates/e2e/setup/project.js
Comment (wschurman) [inline packages/expo-updates/e2e/setup/project.js:558]: prefer named arguments for booleans (so it's not a naked `true`/`false` at callsite)
```suggestion
async function setupE2EAppAsync(projectRoot, localCliBin, repoRoot, { isTV } = { isTV: false }) {
```
(I haven't written JS in a long time so I have no idea if this is valid syntax)

File: packages/expo-updates/e2e/setup/project.js
Comment (wschurman) [inline packages/expo-updates/e2e/setup/project.js:103]: same as other comment about named boolean args

File: packages/expo-updates/e2e/setup/create-updates-test.js
Comment (wschurman) [inline packages/expo-updates/e2e/setup/create-updates-test.js:25]: revert?

File: packages/expo-updates/e2e/setup/project.js
Comment (wschurman) [inline packages/expo-updates/e2e/setup/project.js:173]: don't think you need the `rm` as `writeFile` overwrites by default

File: packages/expo-updates/e2e/setup/project.js
Comment (wschurman) [inline packages/expo-updates/e2e/setup/project.js:368]: revert?

**Review by wschurman — COMMENTED** [review]
This looks fine to me. Couple inline comments. Plus a few things to help get this landed:
- Would be good to reduce the number of reviewers on this to 2-3 at most. It's probably suffering from "someone else will review" syndrome.
- Would be good to have someone else review the template as well. How did you generate the template? (if a copy of `expo-template-bare-minimum` can you detail which changes you made so reviewers don't have to read all the code)

File: packages/expo-updates/e2e/setup/project.js
Comment (douglowder) [reply packages/expo-updates/e2e/setup/project.js:558]: Sure I can rewrite the function args, either syntax is fine in JS

File: packages/expo-updates/e2e/setup/create-updates-test.js
Comment (douglowder) [reply packages/expo-updates/e2e/setup/create-updates-test.js:25]: Actually we should probably move to hermes by default for testing, as JSC probably won't support all features in future. I'll remove the comment.

File: packages/expo-updates/e2e/setup/project.js
Comment (douglowder) [reply packages/expo-updates/e2e/setup/project.js:173]: This is me being cautious... This pattern is used elsewhere in the scripts.

File: packages/expo-updates/e2e/setup/project.js
Comment (douglowder) [reply packages/expo-updates/e2e/setup/project.js:368]: Same comment as above.

**Comment by douglowder** [comment]
@wschurman :

> * Would be good to reduce the number of reviewers on this to 2-3 at most. It's probably suffering from "someone else will review" syndrome.

Good idea, done.

> * Would be good to have someone else review the template as well. How did you generate the template? (if a copy of `expo-template-bare-minimum` can you detail which changes you made so reviewers don't have to read all the code)

Hopefully @brentvatne or @byCedric can review the template. It is essentially `expo-template-blank-typescript`, with native support files added from `expo-template-bare-minimum`. The main native file changes were (1) changes to Xcode project to have it target tvOS instead of iOS, (2) modify SplashScreen storyboard to work on TV instead of phone, and (3) modify Info.plist for TV.

**Review by wschurman — APPROVED** [review]
Approving. Would be good to get one more review since I have no experience with TV.

**Comment by yusufyildirim** [comment]
Is there a plan to make expo-dev-client TV friendly as a part of this effort? I believe the dev-client UI's not focus navigation compatible (or at least friendly) right now if i remember right. Referring to my prior tests on Android, have no idea how it'd work on tvOS.

**Comment by douglowder** [comment]
@yusufyildirim we hope to make dev client work, but we are still scoping the effort needed.

**Comment by yusufyildirim** [comment]
Thanks for the info @douglowder. Incredible work! I'm looking forward to start using Expo for TV platforms.

**Comment by douglowder** [comment]
Since the special TV version of react-native will not pass the usual version check that happens in `npx expo install --check` or `npx expo-doctor`, adding @brentvatne 's suggested config to `package.json` in the template to exclude it from the check.

**Review by tsapeta — COMMENTED** [review]
How much the TV template differs from the bare-minimum template? Maybe we can use bare-minimum and apply some config plugins / codemods on it? Wouldn't it be easier to maintain?
That's something I was thinking to do for macOS support.

**Comment by douglowder** [comment]
@tsapeta see my response to Will above. It is a Typescript template unlike bare-minimum.

I'm not sure it's good to have the modifications done in a config plugin, but it might be more maintainable to have the template be generated in a "prepare" script that copies from the other templates and makes the right modifications.

**Comment by tsapeta** [comment]
Oh ok, thanks @douglowder for the explanation 👍

**Comment by douglowder** [comment]
Yes indeed 😃 It is the Typescript template plus all the native files from bare minimum, so that a new TV project would be Typescript by default (just like the RN core template is).

**Comment by tsapeta** [comment]
We were considering to convert the bare minimum template to use TS (and generally remove non-TS templates), so I think it still makes sense to do what I said before 🤔

**Comment by douglowder** [comment]
@tsapeta thanks for the feedback! What I'd like to do is go ahead and merge this, so we can publish an alpha release and do some testing -- then work on dynamically generating the template in a separate PR.

**Review by tsapeta — APPROVED** [review]
Cool, your plan sounds good to me 👍

**Comment by sunzongzheng** [comment]
I used this command to create the project
`npx create-expo-app my-app --template expo-template-tv`

and then executed `yarn ios` and reported this error.
```shell
Command `pod install` failed.
└─ Cause: The platform of the target `myapp` (tvOS 13.0) is not compatible with `EXConstants (15.0.0)`, which does not support `tvOS`
```

So I would like to know what is the current correct command to use this tv template

**Pull request merged**
